File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // puntero 1
        for(int i = 0; i < nums.size(); i++){
            //puntero 2
            for(int x = nums.size() - 1; x > 0; x--){
                if(nums[i] + nums[x] == target && i != x){
                    vector<int> result = {i, x};
                    return result;
                }
            }
        }
    }
};

File: Data Structures & Algorithms/binary-search/submission-2.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1

        while(start <= end):
            # get middle
            middle = (start + end)//2
            
            if target == nums[middle]:
                return middle
            
            if target > nums[middle]:
                start = middle + 1
            
            if target < nums[middle]:
                end = middle - 1
            
        return -1